### DISCUSSION TITLE
Map different source attributes to properties of event

### DISCUSSION THREAD

**Discussion by javier-molina** [derekantrican/GAS-ICS-Sync#434] [Ideas]
I'd like to have the ability to perform a particular action or mapping based on the source event properties. The case is this:

I create an event in my outlook calendar at work to block some time. I want that event to either be skipped by GAS-ICS-Sync when imported to Google calendar or be set as Free.

The source event in Outlook can't be set as Free otherwise people at work will see me as available. I have thought that if GAS-ICS-Sync had a simple way to map say Out of office status (in outlook) to Free in Google it would do all that I need.

My programming skills are a little bit rust but if at least someone can point me to the object in GAS-ICS-Sync where I can read the outlook properties and the object in Google calendar where I need to change the availability I could give it a go.

Chances are that someone has a different way to solve this problem already.

**Comment by jonas0b1011001** [comment]
For Skipping/Filtering you can have a look at #376

Properties are mapped here
https://github.com/derekantrican/GAS-ICS-Sync/blob/721716039b473929861139de557277571b7af9d6/Helpers.gs#L344

It will probably be helpful to download your source ics and open it in a text editor to get familiar with the ical syntax, #384 might be helpful aswell.

**Comment by javier-molina** [comment]
Awesome, thanks I will continue the conversation on #376
